src/query/functions-v2/src/scalars/comparison.rs

```diff
@@ -0,0 +1,456 @@
+                            );
+                        registry
+                            .register_2_arg::<NumberType<NUM_TYPE>, NumberType<NUM_TYPE>, BooleanType, _, _>(
+                                "lt",
+                                FunctionProperty::default(),
+                                |_, _| None,
+                                |lhs, rhs, _| lhs < rhs,
+                            );
+                        registry
+                            .register_2_arg::<NumberType<NUM_TYPE>, NumberType<NUM_TYPE>, BooleanType, _, _>(
+                                "lte",
+                                FunctionProperty::default(),
+                                |_, _| None,
+                                |lhs, rhs, _| lhs <= rhs,
+                            );
+                    }
+                });
+            }
+            _ => todo!(),
+        });
+    }
+
+    // TODO: wait @b41sh complete jsonb cmp, and need support
+    // `VariantType like VariantType`
+    // `VariantType regexp VariantType`
+    // registry.register_2_arg::<VariantType, VariantType, BooleanType, _, _>(
+    //     "eq",
+    //     FunctionProperty::default(),
+    //     |_lhs, _rhs| None,
+    //     |lhs, rhs, _| lhs.cmp(rhs) == Ordering::Equal,
+    // );
+
+    // registry.register_2_arg::<VariantType, VariantType, BooleanType, _, _>(
+    //     "gt",
+    //     FunctionProperty::default(),
+    //     |_lhs, _rhs| None,
+    //     |lhs, rhs, _| lhs.cmp(rhs) == Ordering::Greater,
+    // );
+
+    // registry.register_2_arg::<VariantType, VariantType, BooleanType, _, _>(
+    //     "lt",
+    //     FunctionProperty::default(),
+    //     |_lhs, _rhs| None,
+    //     |lhs, rhs, _| lhs.cmp(rhs) == Ordering::Less,
+    // );
+
+    // registry.register_2_arg::<VariantType, VariantType, BooleanType, _, _>(
+    //     "gte",
+    //     FunctionProperty::default(),
+    //     |_lhs, _rhs| None,
+    //     |lhs, rhs, _| {
+    //         let res = lhs.cmp(rhs);
+    //         res == Ordering::Equal || res == Ordering::Greater
+    //     },
+    // );
+
+    // registry.register_2_arg::<VariantType, VariantType, BooleanType, _, _>(
+    //     "lte",
+    //     FunctionProperty::default(),
+    //     |_lhs, _rhs| None,
+    //     |lhs, rhs, _| {
+    //         let res = lhs.cmp(rhs);
+    //         res == Ordering::Equal || res == Ordering::Less
+    //     },
+    // );
+
+    // registry.register_2_arg::<VariantType, VariantType, BooleanType, _, _>(
+    //     "noteq",
+    //     FunctionProperty::default(),
+    //     |_lhs, _rhs| None,
+    //     |lhs, rhs, _| lhs.cmp(rhs) != Ordering::Equal,
+    // );
+
+    registry.register_passthrough_nullable_2_arg::<StringType, StringType, BooleanType, _, _>(
+        "like",
+        FunctionProperty::default(),
+        |_lhs, _rhs| None,
+        |lhs, rhs, ctx| {
+            let mut map: HashMap<&[u8], Regex> = HashMap::new();
```

Comment:
Try to add the map into functioncontext but the struct functioncontext derived Copy trait. 

HashMap<&[u8], Regex> the Regex do not derived Copy.

So I can add it into functioncontext..

Try to use a closer function but the &[u8] has not enough life cycle...

So use this way to catch the map.

Is there has some way to optimize the function? I think I need some help.

Reply:
how about use `register_function_factory`? like `regexp_instr_fn`

Reply:
> how about use `register_function_factory`? like `regexp_instr_fn`

Yes, that's also a good idea. And I talk with andy, he has another way to resolve this.